### PR TITLE
Adds Handwritten Digit Recognition example

### PR DESCRIPTION
**Pull request by DnPlas**
Adding the Handwritten Digit Recognition based on DLRS, Pytorch and MNIST.
This series introduces the following changes:
* Creates kubeflow/ in root directory with files for running the mentioned example on kubeflow pipelines
* Creates handwritten_digit_recog/ in root directory with files for running the mentioned example on Docker containers

### CODE REVIEW

File: handwritten_digit_recog/python/train.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        self.dropout1 = nn.Dropout2d(0.25)
+        self.dropout2 = nn.Dropout2d(0.5)
+        self.fc1 = nn.Linear(9216, 128)
+        self.fc2 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = self.conv2(x)
+        x = F.max_pool2d(x, 2)
```
Comment (unrahul) [inline handwritten_digit_recog/python/train.py]: is there are reason why no activation fn is given here?

File: handwritten_digit_recog/python/train.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    )
+    parser.add_argument(
+        "--epochs",
+        type=int,
+        default=1,
+        metavar="N",
+        help="number of epochs to train (default: 10)",
+    )
+    parser.add_argument(
+        "--lr", type=float, default=1.0, metavar="LR", help="learning rate (default: 1.0)"
```
Comment (unrahul) [inline handwritten_digit_recog/python/train.py]: too big a learning rate, let the default be something like 0.01?

File: handwritten_digit_recog/python/train.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        default=10,
+        metavar="N",
+        help="how many batches to wait before logging training status",
+    )
+
+    args = parser.parse_args()
+
+    torch.manual_seed(args.seed)
+
+    kwargs = {"num_workers": 1, "pin_memory": True}
```
Comment (unrahul) [inline handwritten_digit_recog/python/train.py]: do a multiprocessing.cpu_count and let the num_workers be half of physical CPU?

Comment (DnPlas) [reply handwritten_digit_recog/python/train.py]: Agree

File: handwritten_digit_recog/python/train.py (hunk 2 above)
Comment (DnPlas) [reply handwritten_digit_recog/python/train.py]: Agree

File: handwritten_digit_recog/python/train.py (hunk 1 above)
Comment (unrahul) [reply handwritten_digit_recog/python/train.py]: Hey @DnPlas does this network perform well, without a non linear activation here?

File: handwritten_digit_recog/python/train.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    )
+    parser.add_argument(
+        "--epochs",
+        type=int,
+        default=1,
+        metavar="N",
+        help="number of epochs to train (default: 10)",
+    )
+    parser.add_argument(
+        "--lr", type=float, default=1.0, metavar="LR", help="learning rate (default: 0.01)"
```
Comment (unrahul) [inline handwritten_digit_recog/python/train.py]: nit: change default=0.01 :)

File: handwritten_digit_recog/python/train.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        "--batch-size",
+        type=int,
+        default=64,
+        metavar="N",
+        help="input batch size for training (default: 64)",
+    )
+    parser.add_argument(
+        "--test-batch-size",
+        type=int,
+        default=1000,
```
Comment (unrahul) [inline handwritten_digit_recog/python/train.py]: may be make it same as the training batch, 64 ?
